src/slicer.h

```diff
@@ -109,12 +116,350 @@ class SlicerLayer
 
     /*!
      * Try to close up polylines into polygons while they have large gaps in them.
-     * 
+     *
      * Clears all open polylines which are used up in the process
-     * 
+     *
      * \param[in,out] open_polylines The polylines which are stiched, but couldn't be closed into a loop yet
      */
     void stitch_extensive(Polygons& open_polylines);
+
+private:
+    /*!
+     * \brief This class represents the location of an end point of a
+     * polyline in a polyline vector.
+     *
+     * The location records the index in the polyline vector and
+     * whether this is the vertex at the start of the polyline or the
+     * vertex at the end.
+     */
+    class Terminus
+    {
+    public:
+        /*! A representation of Terminus that can be used as an array index. */
```

Comment:
I'll see if I can make it a little bit clearer.  I'm having trouble coming up with good descriptions for some of the concepts even though they are clear in my head.  I think it is because I am thinking of them spatially and so don't have a ready made verbal representation.